src/lsm/table_mutable.zig

```diff
@@ -42,7 +39,142 @@ pub fn TableMutableType(comptime Table: type, comptime tree_name: [:0]const u8)
+                    return compare_keys(ctx.key_from_index(a), ctx.key_from_index(b)) == .eq;
+                }
+
+                pub inline fn hash(ctx: IndexMapContext, index: u32) u64 {
+                    const key = ctx.key_from_index(index);
+                    return std.hash.Wyhash.hash(0, mem.asBytes(&key));
+                }
+            };
+
+            index_map: IndexMap,
+            values: *Values,
+            active: u32,
+
+            pub fn init(allocator: mem.Allocator) !ValuesMap {
+                var index_map = IndexMap{};
+                try index_map.ensureTotalCapacityContext(allocator, value_count_max, undefined);
+                errdefer index_map.deinit(allocator);
+
+                const values = try allocator.create(Values);
+                errdefer allocator.destroy(values);
+
+                return ValuesMap{
+                    .index_map = index_map,
+                    .values = values,
+                    .active = 0,
+                };
+            }
+
+            pub fn deinit(map: *ValuesMap, allocator: mem.Allocator) void {
+                allocator.destroy(map.values);
+                map.index_map.deinit(allocator);
+            }
+
+            pub inline fn count(map: *const ValuesMap) u32 {
+                return map.index_map.count();
+            }
+
+            pub inline fn clear(map: *ValuesMap) void {
+                map.index_map.clearRetainingCapacity();
+                map.active = 0;
+            }
+
+            pub fn find(map: *const ValuesMap, key: Key) ?*const Value {
+                const ctx = IndexMapContext{ .values = map.values, .search_key = key };
+                const index = map.index_map.getKeyAdapted(ctx.search_index, ctx) orelse return null;
+                return &map.values.items[index];
+            }
+
+            pub fn upsert(map: *ValuesMap, value: *const Value) *Value {
+                const ctx = IndexMapContext{ .values = map.values, .search_key = key_from_value(value) };
+                const result = map.index_map.getOrPutAssumeCapacityContext(ctx.search_index, ctx);
+                if (result.found_existing) {
+                    return &map.values.items[result.key_ptr.*];
+                }
+
+                const index = @intCast(u32, map.active);
+                map.active += 1;
+
+                result.key_ptr.* = index;
+                map.values.active[index / 8] |= @as(u8, 1) << @intCast(u3, index % 8);
+                return &map.values.items[index];
+            }
+
+            pub fn fetch_remove(map: *ValuesMap, value: *const Value) ?*Value {
+                const ctx = IndexMapContext{ .values = map.values, .search_key = key_from_value(value) };
+                const kv = map.index_map.fetchRemoveContext(ctx.search_index, ctx) orelse return null;
+                const index = kv.key;
+
+                map.values.active[index / 8] &= ~(@as(u8, 1) << @intCast(u3, index % 8));
```

Comment:
```zig
[index / 8]
```

As per TigerStyle, let's always use `@divExact()` etc. to indicate what we intend precisely with respect to flooring.

Reply:
Also, is there anything we expect of the existing value for `active[index / 8]` here?

Or do we intend to unilaterally `AND` it regardless of the existing value?

Reply:
Is there anything we need to still do with respect to `map.active`?

Reply:
We expect it's bit to already be set. Ill add that.